src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -902,73 +900,61 @@ void AdvertiserMinMdns::AdvertiseRecords(BroadcastAdvertiseType type)
-        {
-            if (!ShouldAdvertiseOn(interfaceId, ipAddress))
-            {
-                continue;
-            }
+        packetInfo.Clear();
 
-            chip::Inet::IPPacketInfo packetInfo;
+        // Attempt to get a "Real" IP address since `Type()` on a broadcast address
+        // always returns IPAddressType::kAny
+        CHIP_ERROR err = GetFirstIpAddress(interfaceId, addressType, packetInfo.SrcAddress);
 
-            packetInfo.Clear();
-            packetInfo.SrcAddress = ipAddress;
-            if (ipAddress.IsIPv4())
-            {
-                BroadcastIpAddresses::GetIpv4Into(packetInfo.DestAddress);
-            }
-            else
-            {
-                BroadcastIpAddresses::GetIpv6Into(packetInfo.DestAddress);
-            }
-            packetInfo.SrcPort   = kMdnsPort;
-            packetInfo.DestPort  = kMdnsPort;
-            packetInfo.Interface = interfaceId;
-
-            // Advertise all records
-            //
-            // TODO: Consider advertising delta changes.
-            //
-            // Current advertisement does not have a concept of "delta" to only
-            // advertise changes. Current implementation is to always
-            //    1. advertise TTL=0 (clear all caches)
-            //    2. advertise available records (with longer TTL)
-            //
-            // It would be nice if we could selectively advertise what changes, like
-            // send TTL=0 for anything removed/about to be removed (and only those),
-            // then only advertise new items added.
-            //
-            // This optimization likely will take more logic and state storage, so
-            // for now it is not done.
-            QueryData queryData(QType::PTR, QClass::IN, false /* unicast */);
-            queryData.SetIsInternalBroadcast(true);
-
-            for (auto & it : mOperationalResponders)
-            {
-                it.GetAllocator()->GetQueryResponder()->ClearBroadcastThrottle();
-            }
-            mQueryResponderAllocatorCommissionable.GetQueryResponder()->ClearBroadcastThrottle();
-            mQueryResponderAllocatorCommissioner.GetQueryResponder()->ClearBroadcastThrottle();
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Discovery, "Failed to find a suitable IP address to advertise: %" CHIP_ERROR_FORMAT, err.Format());
+            continue;
+        }
 
-            CHIP_ERROR err = mResponseSender.Respond(0, queryData, &packetInfo, responseConfiguration);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(Discovery, "Failed to advertise records: %" CHIP_ERROR_FORMAT, err.Format());
-            }
+        if (addressType != chip::Inet::IPAddressType::kIPv6)
+        {
+            BroadcastIpAddresses::GetIpv4Into(packetInfo.DestAddress);
+        }
+        else
+        {
+            BroadcastIpAddresses::GetIpv6Into(packetInfo.DestAddress);
+        }
```

Comment:
I don't understand this. It appears there is some kind of either-or related to IPv6. We should announce with IPv6 always, and never only with IPv4. Optionally doing for IPv4 should be a configuraiton, not a fallback.

Reply:
The reason for the awkward condition flip is because `kIPv4` is not defined if IPV4 is not enabled, so I always compare with IPv6. Logic is the same.

Besides using `!= kIPv6` instead of `== kIPv4` the seems the same (it assume any and unknown is not used). This says "for IPv4 replies use IPv4 destination, for IPv6 use IPv6.